worlds/hk/test/test_grub_count.py

```diff
@@ -0,0 +1,110 @@
+from . import linkedTestHK, WorldTestBase
+from Options import ItemLinks
+
+
+class test_grubcount_limited(linkedTestHK, WorldTestBase):
+    options = {
+        "RandomizeGrubs": True,
+        "GrubHuntGoal": 20,
+        "Goal": "any",
+    }
+    item_link_group = [{
+        "name": "ItemLinkTest",
+        "item_pool": ["Grub"],
+        "link_replacement": True,
+        "replacement_item": "Grub",
+    }]
+    expected_grubs = 20
+
+
+class test_grubcount_default(linkedTestHK, WorldTestBase):
+    options = {
+        "RandomizeGrubs": True,
+        "Goal": "any",
+    }
+    item_link_group = [{
+        "name": "ItemLinkTest",
+        "item_pool": ["Grub"],
+        "link_replacement": True,
+        "replacement_item": "Grub",
+    }]
+    expected_grubs = 46
+
+
+class test_grubcount_all_unlinked(linkedTestHK, WorldTestBase):
+    options = {
+        "RandomizeGrubs": True,
+        "GrubHuntGoal": "all",
+        "Goal": "any",
+    }
+    item_link_group = []
+    expected_grubs = 46
+
+
+class test_grubcount_all_linked(linkedTestHK, WorldTestBase):
+    options = {
+        "RandomizeGrubs": True,
+        "GrubHuntGoal": "all",
+        "Goal": "any",
+    }
+    item_link_group = [{
+        "name": "ItemLinkTest",
+        "item_pool": ["Grub"],
+        "link_replacement": True,
+        "replacement_item": "Grub",
+    }]
+    expected_grubs = 46 + 23
```

Comment:
I asked in DMs, but I should put this here as well:
Why is the expected grub count 69 when there are 92 Grubs in the pool and playthrough also expects players to get 92 Grubs?

Reply:
Apparently only *one* of the players is linking the replacement Grub. I hate itemlinks

Reply:
Yes. There are always 1 fewer replacement items, per linked item, than the number of items being linked together.

If there were 6 players in an item link, linking an item between them would become 1 linked item + 5 replacements.

Reply:
Oh, what I said is also apparently completely wrong. The link_relacement is for the group, the *replacement item* is per player